pkg/cli/clients/clients.go

```diff
@@ -137,40 +137,74 @@ type LogStream struct {
+	// DeleteApplication deletes an application and all of its resources by its name (or id).
+	DeleteApplication(ctx context.Context, applicationNameOrID string) (bool, error)
 
-	// ListEnvironmentsInResourceGroup lists all environments in the configured scope (assumes configured scope is a resource group)
-	ListEnvironmentsInResourceGroup(ctx context.Context) ([]corerp.EnvironmentResource, error)
+	// ListEnvironments lists all environments in the configured scope (assumes configured scope is a resource group).
+	ListEnvironments(ctx context.Context) ([]corerp.EnvironmentResource, error)
 
 	// ListEnvironmentsAll lists all environments across resource groups.
 	ListEnvironmentsAll(ctx context.Context) ([]corerp.EnvironmentResource, error)
-	GetEnvDetails(ctx context.Context, envName string) (corerp.EnvironmentResource, error)
-	DeleteEnv(ctx context.Context, envName string) (bool, error)
-	CreateUCPGroup(ctx context.Context, planeName string, resourceGroupName string, resourceGroup ucp_v20231001preview.ResourceGroupResource) error
-	DeleteUCPGroup(ctx context.Context, planeName string, resourceGroupName string) (bool, error)
-	ShowUCPGroup(ctx context.Context, planeName string, resourceGroupName string) (ucp_v20231001preview.ResourceGroupResource, error)
-	ListUCPGroup(ctx context.Context, planeName string) ([]ucp_v20231001preview.ResourceGroupResource, error)
-
-	// ShowRecipe shows recipe details including list of all parameters for a given recipe registered to an environment
-	ShowRecipe(ctx context.Context, environmentName string, recipe corerp.RecipeGetMetadata) (corerp.RecipeGetMetadataResponse, error)
+
+	// GetEnvironment retrieves an environment by its name (in the configured scope) or resource ID.
+	GetEnvironment(ctx context.Context, environmentNameOrID string) (corerp.EnvironmentResource, error)
+
+	// GetRecipeMetadata shows recipe details including list of all parameters for a given recipe registered to an environment.
+	GetRecipeMetadata(ctx context.Context, environmentNameOrID string, recipe corerp.RecipeGetMetadata) (corerp.RecipeGetMetadataResponse, error)
+
+	// CreateOrUpdateEnvironment creates an environment by its name (or id).
+	CreateOrUpdateEnvironment(ctx context.Context, environmentNameOrID string, resource *corerp.EnvironmentResource) error
+
+	// DeleteEnvironment deletes an environment and all of its resources by its name (in the configured scope) or resource ID.
+	DeleteEnvironment(ctx context.Context, environmentNameOrID string) (bool, error)
+
+	// ListResourceGroups lists all resource groups in the configured scope.
+	ListResourceGroups(ctx context.Context, planeName string) ([]ucp_v20231001preview.ResourceGroupResource, error)
+
+	// GetResourceGroup retrieves a resource group by its name.
+	GetResourceGroup(ctx context.Context, planeName string, resourceGroupName string) (ucp_v20231001preview.ResourceGroupResource, error)
```

Comment:
Are the resourcegroup APIs not with NameOrID for a reason?

Reply:
I thought about this decided against. Right now they take the `planeName` and the `resourceGroupName` which means that they can always build the resource id fully. So taking in a `resourceGroupNameOrId` would make the plane name redundant.

Reply:
Does this sound right? any better ideas?

Reply:
This is fine too but I feel it would be better if all APIs accept similar input formats.

Reply:
What do you think the *right* signature is?

An API like `GetEnvironment(ctx, nameOrId)` works because the client has a notion of "default scope" that can be prepended to the name.

What do you think the behavior of `GetResourceGroup(ctx, nameOrId)` should be?